lib/autoload/polyglot/language.rb

```diff
@@ -0,0 +1,90 @@
+# An abstract language. This is an ActiveRecord class for other models to reference to a family of
+# languages.
+#
+# Do *NOT* remove languages after they have been defined because a database record corresponds to
+# a class implemented here.
+#
+# Each language acts as a singleton class, call +Polyglot::Language.instance+ to retrieve the
+# instance for the language.
```

Comment:
@allenwq I've rewritten this documentation, see if this is clearer:
```ruby
# An abstract language. This is an ActiveRecord class for other models to reference to a family of
# languages. Languages become concrete and can be referenced in the database when that language
# implements +Polyglot::Language.instance+.
#
# In addition as being an entity which can be referenced in the database, this also adds the
# scripts and stylesheets needed to syntax highlight code in languages defined here.
#
# Each subclass represents a language ancestry, such as differing language versions (see the Python
# language definition.) Each concrete language has a unique +Polyglot::Language.instance+ which
# is internally called the +root+ of the language. Derived languages can be defined at runtime to
# utilise the syntax highlighting capabilities of the root language, while requiring a separate
# runtime environment to run programs written in the derived language.
#
# Do *NOT* remove languages after they have been defined because a database record corresponds to
# a class implemented here.
```